### PR TITLE
Add few more test for dex.

### PR DESCRIPTION
**Pull request by vivekvpandya**
Add few more test for dex.
Add extrinsics for add/remove_liquidity().
Update benchmarks.

### CODE REVIEW

File: frame/curve-amm/src/tests.rs [hunk 1]
```diff
@@ -211,6 +211,64 @@ fn add_lp_imbalanced() {
...
+			&BOB,
+			pool_id,
+			bob_usdc,
+			bob_usdt,
+			expected_min_value,
+			false
+		));
+		let lp = Tokens::balance(pool.lp_token, &BOB);
+		// must have received some lp tokens
+		assert!(lp > 0_u128);
```
Comment (KaiserKarel) [inline frame/curve-amm/src/tests.rs:250]: Can we assert that we received the correct amount?

File: frame/uniswap-v2/src/lib.rs [hunk 2]
```diff
@@ -302,6 +302,46 @@ pub mod pallet {
...
+		pub fn add_liquidity(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			base_amount: T::Balance,
+			quote_amount: T::Balance,
+			min_mint_amount: T::Balance,
+			keep_alive: bool,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let _ = <Self as CurveAmm>::add_liquidity(
```
Comment (KaiserKarel) [inline frame/uniswap-v2/src/lib.rs:316]: Why CurveAmm? Should we rename the trait to just AMM?

**Review by KaiserKarel — CHANGES_REQUESTED** [review]
Would be good to check that the obtained LP amount is correct

File: frame/uniswap-v2/src/lib.rs (hunk 2 above)
Comment (KaiserKarel) [reply frame/uniswap-v2/src/lib.rs:316]: Probably good for a second PR

**Review by hussein-aitlahcen — DISMISSED** [review]
Thanks for adding a bunch of tests, next step is to add some proptests.

File: frame/curve-amm/src/benchmarking.rs [hunk 3]
```diff
@@ -19,6 +22,57 @@ benchmarks! {
...
+			  &owner,
+			  pool_id,
+			  initial_usdc,
+			  initial_usdt,
+			  0.into(),
+			  false
+	  ));
+	  let pool_info = StableSwap::<T>::get_pool(pool_id).expect("impossible; qed;");
+	  let lp_amount = T::Assets::balance(pool_info.lp_token, &owner);
+  }: _(RawOrigin::Signed(owner), pool_id, lp_amount, (0_u128).into(), (0_u128).into())
```
Comment (dzmitry-lahoda) [inline frame/curve-amm/src/benchmarking.rs:74]: does this test handles error to remove? so if remove will fail, benchmark with fail?

File: frame/curve-amm/src/benchmarking.rs [hunk 4]
```diff
@@ -19,6 +22,57 @@ benchmarks! {
...
+		  Permill::from_percent(1),
+	  ) .expect("impossible; qed;");
+	  let unit = 1_000_000_000_000;
+	  // 100_000_000 USDC , 100_000_000 USDT
+	  let initial_usdc: T::Balance = (100_000_000_u128 * unit).into();
+	  let initial_usdt: T::Balance = (100_000_000_u128 * unit).into();
+	  // Mint the tokens
+	  assert_ok!(T::Assets::mint_into(usdc, &owner, initial_usdc));
+	  assert_ok!(T::Assets::mint_into(usdt, &owner, initial_usdt));
+  }: _(RawOrigin::Signed(owner), pool_id, initial_usdc, initial_usdt, 0.into(), false)
```
Comment (dzmitry-lahoda) [inline frame/curve-amm/src/benchmarking.rs:43]: does this test handles error case?

File: frame/curve-amm/src/tests.rs [hunk 5]
```diff
@@ -217,6 +232,64 @@ fn add_lp_imbalanced() {
...
+			bob_usdc,
+			bob_usdt,
+			expected_min_value,
+			false
+		));
+		let lp = Tokens::balance(pool.lp_token, &BOB);
+		// must have received some lp tokens
+		assert!(lp == bob_usdc + bob_usdt);
+
+		let bob_usdc = 1000 * unit;
```
Comment (dzmitry-lahoda) [inline frame/curve-amm/src/tests.rs:273]: may be split into 2 tests?

File: frame/curve-amm/src/tests.rs [hunk 6]
```diff
@@ -242,14 +315,80 @@ fn remove_lp_failure() {
...
+				lp,
+				min_expected_usdc,
+				min_expected_usdt
+			),
+			crate::Error::<Test>::CannotRespectMinimumRequested
+		);
+	});
+}
+
+//
```
Comment (dzmitry-lahoda) [inline frame/curve-amm/src/tests.rs:346]: // non needed
// -> ///

File: frame/uniswap-v2/src/lib.rs [hunk 7]
```diff
@@ -302,9 +302,55 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Add liquidity to a stable-swap pool.
```
Comment (dzmitry-lahoda) [inline frame/uniswap-v2/src/lib.rs:306]: wrong comment

**Review by KaiserKarel — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Pull request merged**